src/functions-and-arrays.js

```diff
@@ -1,41 +1,172 @@
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(sum) {
+  if (!sum)        return;
+  let long = sum.length;
+  let soma = 0;
+  for (let i = 0; i < long; i++){
+      soma += sum[i];
+  }
+  return soma;
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+const newArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+
+function sum(arr) {
+  if (arr.length === 0) return 0;
+  if (arr.length === 1) return arr[0];
+  let checkZero = true;
+  let checkNum = true;
+  let checkString = true;
+  let checkMixNs = true;
+  let soma = 0;
+  for (let i = 0; i < arr.length; i++){
+    if (arr[i] !== 0){
+      checkZero = false;
+      break
+    }
+  } 
+  if (checkZero === true){
+    return 0;
+  }
+  // checking if it is an array of numbers
+  for (let i = 0; i < arr.length; i++){
+    if (typeof arr[i] === 'number'){
+      soma += arr[i];
+    } else {
+      checkNum = false;
+      break;
+    }
+  }
+  if (checkNum === true){
+    return soma;
+  } 
+  // checking if it is an array of strings
+  for (let i = 0; i < arr.length; i++){
+    if (typeof arr[i] === 'string'){
+      soma += arr[i].length;
+    } else {
+      checkString = false;
+      break;
+    }
+  }
+  if (checkString === true){
+    return soma;
+  } 
+// check if it is a mixed array of strings and numbers
+  for (let e = 0; e < arr.length; e++){
+    if (typeof arr[e] === 'string'){
+      soma += arr[e].length;
+    } else if (typeof arr[e] === 'number'){
+      soma += arr[e];
+    } else {
+      checkMixNs = false;
+      break;
+    }
+  }
+  if (checkMixNs === true){
+    return soma;
+  } 
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(mediaNum){
+  if (mediaNum.length === 0) return null;
+
+  let long = mediaNum.length;
+  let media = 0;
+  for (let i = 0; i < long; i++){
+      media += mediaNum[i];
+  }
+  return media / long;
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(avgNum) {
+  if (avgNum.length === 0) return null;
+  if (avgNum.length === 1) return avgNum[0].length;
+  let long = avgNum.length;
+  let media = 0;
+  for (let i = 0; i < long; i++){
+      media += avgNum[i].length;
+  }
+  return media / long;
+ }
 
 // Bonus - Iteration #4.1
-function avg() {}
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+
+function avg(array) {
+  if (array.length === 0) return null;
+  arrayAvg = 0;
+  for (let i = 0; i < array.length; i++){
+    if(typeof array[i] === "number"){
+      arrayAvg += array[i];
+    } else if (typeof array[i] === "string"){
+      arrayAvg += array[i].length;
+    } else {
+      arrayAvg += 1;
+    }
+  }
+  arrayAvg /= 10;
+  return arrayAvg;
```

Comment:
Here, the idea was to return the average of the array.
You can use the sum() function from before for that.

`return sum(array)/array.length;`